src/reporters.ts

```diff
@@ -201,6 +202,66 @@ export const reportInvalidRelation = ({ markers, lines, lineIndex, value, validR
   }
 };
 
+export const reportInvalidTypeRelation = ({
+  markers,
+  lines,
+  lineIndex,
+  value,
+  typeName,
+  relationName,
+}: ReporterOpts) => {
+  reportError({
+    markers,
+    lines,
+    lineIndex,
+    value,
+    message: `\`${relationName}\` is not a valid relation for \`${typeName}\`.`,
+    relatedInformation: { type: "invalid-relation-type", relation: relationName, typeName: typeName } as any,
+  });
+};
+
+export const reportInvalidType = ({ markers, lines, lineIndex, value, typeName }: ReporterOpts) => {
+  reportError({
+    markers,
+    lines,
+    lineIndex,
+    value,
+    message: `\`${typeName}\` is not a valid type.`,
+    relatedInformation: { type: "invalid-relation-type", typeName: typeName } as any,
+  });
+};
+
+export const noEntryPoint = ({ markers, lines, lineIndex, value, validRelations }: ReporterOpts) => {
+  const data = getValidRelationsArray(validRelations);
+  const isInValid = !data?.includes(value);
+  if (isInValid) {
+    reportError({
+      markers,
+      lines,
+      lineIndex,
+      value,
+      message: `\`${value}\` is an impossible relation (no entrypoint).`,
```

Comment:
<img width="606" alt="image" src="https://user-images.githubusercontent.com/5374887/198162548-1cc6fab7-b47f-477d-8648-78c7dd8b82fc.png">

```suggestion
      message: `\`${value}\` cannot be defined as a circular relation.`,
```

Reply:
Maybe we can clean up the actual error message in separate PR?